sys/parsoid.js

```diff
@@ -344,7 +392,9 @@ class ParsoidService {
                         body: res.body[format].body
                     };
                     resp.headers.etag = mwUtil.makeETag(rp.revision, tid);
-                    return this.saveParsoidResult(hyper, req, format, tid, res)
+                    return this.saveParsoidResultToLatest(hyper, req, tid, res)
+                    .catch({ status: 412 }, () =>
```

Comment:
Thanks for the pointer. I was somehow assuming that this was hitting the table storage backend, and didn't find the 412 there.

Regarding the interface, I agree that 412 kind of makes sense here. It just seems surprising without making a conditional request. The condition we are after is "if rev > previous rev". None of the existing conditional request headers fit that squarely, but perhaps it would still be clearer to  communicate the condition in the request. The best I can come up with using existing conditions would be `if-none-match: W/"12345"`: Only store if existing content does not *weakly* match rev 12345. I guess another option would be to create a new header, for example `if-version-lt: 12345`.

For some of these more specialized buckets we could also consider calling them directly as modules, rather than making HTTP calls to them through the router. This might be a bit easier to understand, and would avoid some routing overhead. Main loss would be less detailed metrics about internal bucket requests.